### ISSUE TITLE
New Feature: `onyo set` should be able to set the `onyo.path.parent` pseudo key

### ISSUE THREAD

**Issue by aqw** [psyinfra/onyo#693]
In effect, this is an `onyo mv` via `onyo set`.

Currently, the `directory` pseudo key (soon to be `onyo.path.parent`; see #688) is queryable with `get` and setable with `onyo new --keys` and the TSV file. It feels inconsistent to all the key to be set in some places and not another.

If parent directories are missing, they should be created automatically. This matches the behavior described in #655 (which ironically I don't particularly like  ;-)

**Comment by TobiasKadelka** [comment]
While setting `onyo.path.parent` is probably more common and addresses the directory in which e.g. an asset is, I want to add that setting other pseudo keys related to `onyo.path` should also work.

A source or destination for a move can obviously be the full path of an asset, as long as it is i.e. `onyo.path=<new path together with asset name>`.

> If parent directories are missing, they should be created automatically. This matches the behavior described in https://github.com/psyinfra/onyo/issues/655 (which ironically I don't particularly like ;-)

I am happy to hear that you see the irony yourself ;) And I continue to think that `onyo mv` should, in the same way `onyo set` does in your description, be able to create a directory whenever a user specifies in a inventory system that this is the new location for a piece of hardware. (Nowadays even more so with the "Inventory Operations" always being printed :)

In summary: I agree with this issue and want the feature described to be added, too! 👍

**Comment by aqw** [comment]
>  I want to add that setting other pseudo keys related to `onyo.path` should also work.

My initial response was "not no but hell no", but then I remembered that we want to support templates and directories in `onyo get` (#674 #675).

If (big if) we extend that to `set`, then I do see use cases for setting `onyo.path.relative` and `onyo.path.file`.

However, that is adding a lot of complexity to `set` and `unset` for something we don't have use cases for (at least yet).

For assets, setting `onyo.path` keys other than `onyo.path.directory` makes little sense. Setting `onyo.path.relative` cannot allow changing the asset name, which means setting only the directory part of the value, which feels weird. Same thing for `onyo.path.file`.

**Comment by bpoldrack** [comment]
> However, that is adding a lot of complexity to set and unset

That's my concern here. Even if it's limited to `onyo.path.parent` that's already a fundamental change of what `set` does.
Predictably, that's instantly giving rise to more desires about things being settable.

Moreover, since the (awful IMO) #655 is mentioned once more:

I'm not a fan of implicitly creating inventory items (dirs in these cases) with commands that are inherently not about creating stuff. Whether that is `set` or `mv`. Different commands have different purposes and different safeguards/failures.
If I say "new", it's clear that I need to specify things that don't exist yet. If I say "move", "set" or any other notion of "modify something existing" that's just not the case. We shouldn't magically create stuff, when the command is not about creating stuff.
If we are to gown that road, than any notion of unixy+filesystem-based kinda goes straight out of the window and we turn onyo into a DB, where fundamentally all we got is `set`+`get` and everything else is convenience aliases for commonly desired calls to the two. And, of course, we loose half the safe-guards, because everything is a valid call - internal magic will make it so.

**Comment by aqw** [comment]
It seems there's some controversy here. I have changed this from a New Feature issue to a Design issue.
